pyvo/io/uws/endpoint.py

```diff
@@ -9,22 +9,27 @@
 from astropy.io.votable.util import convert_to_writable_filelike
 
 from ...utils.xml.elements import xmlattribute
-from .tree import JobSummary
+from .tree import JobSummary, JobList
 
 __all__ = ["parse_job", "JobFile"]
 
 
-def parse_job(
-    source, pedantic=None, filename=None, _debug_python_based_parser=False
+
+
+def parse_for_object(
```

Comment:
It's a good idea to avoid redundant functions, but it makes the code (or the library) less readable.
Maybe create two functions with `functools.partial`?

Edit: Maybe even put the generic function into `utils.xml`.